lib/ClangImporter/ImportDecl.cpp

```diff
@@ -8138,6 +8138,23 @@ ClangImporter::Implementation::importSwiftAttrAttributes(Decl *MappedDecl) {
                                   /*isUnchecked=*/true);
     }
   }
+
+  // Special handling of `NSNotificationName` static immutable properties.
+  //
+  // These constants could be used with observer APIs from a different isolation
+  // context, so it's more convenient to import them as `nonisolated` unless
+  // they are explicitly isolated to a MainActor.
+  if (!seenMainActorAttr) {
+    auto *DC = MappedDecl->getDeclContext();
+    if (DC->isTypeContext() && isa<VarDecl>(MappedDecl)) {
+      auto *mappedVar = cast<VarDecl>(MappedDecl);
+      if (mappedVar->isStatic() && mappedVar->isLet() &&
+          isNSNotificationName(cast<clang::ValueDecl>(ClangDecl)->getType())) {
```

Comment:
This logic should probably be driven off `importer::isNSNotificationGlobal()` or some version of that. There are *many* constants in the SDK that are declared as type `NSString *`, but are imported as `NSNotification.Name` based on their names having a `Notification` suffix—this logic ought to apply to them, too.

Reply:
Sorry, misunderstood the requirements here. We're trying to fix something that only affects notification names that have been `NS_SWIFT_NAME()`d into an actor-isolated type, and `isNSNotificationGlobal()` rejects anything with an `NS_SWIFT_NAME()`.